src/TextField/EnhancedTextarea.js

```diff
@@ -136,7 +135,9 @@ class EnhancedTextarea extends Component {
     const {prepareStyles} = this.context.muiTheme;
     const styles = getStyles(this.props, this.context, this.state);
     const rootStyles = Object.assign({}, styles.root, style);
-    const textareaStyles = Object.assign({}, styles.textarea, textareaStyle);
+    const textareaStyles = Object.assign({}, styles.textarea, textareaStyle, {
```

Comment:
The switch wouldn't work because this was actually the problem in the first place, the property height: this.state.height was being replaced with height: '100%' of textareaStyle passed in props from the TextField component.
I've made the const styles = getStyles(this.props) change as suggested above and squashed the commits.